src/main/scala/scala/tools/refactoring/implementations/InlineLocal.scala

```diff
@@ -26,10 +26,12 @@ abstract class InlineLocal extends MultiStageRefactoring with ParameterlessRefac
       case None => s.findSelectedOfType[ValDef]
     }
 
+    def isInliningAllowed(sym: Symbol) =
+      (sym.isPrivate || sym.isLocal) && !sym.isMutable && !sym.isValueParameter
+
     selectedValue match {
-      case Some(t) if (t.symbol.isPrivate || t.symbol.isLocal)
-        && !t.symbol.isMutable && !t.symbol.isValueParameter
-        && t.symbol.enclMethod != NoSymbol => Right(t)
```

Comment:
Agreed, that didn't make any sense. Maybe the behavior was different back in 2.8, but I don't really remember.